### PR TITLE
WIP Generic commands 

### PR DESCRIPTION
**Pull request by samiwh**
#### Description
Ability to call generic commnds via the ICommandRequestSender interface.
Like the recent command closures this is a temporary measure.

Needs world commands added

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt [hunk 1]
```diff
@@ -108,13 +110,55 @@ namespace <#= qualifiedNamespace #>
                         return -1;
                     }
 
+                    Action<<#= receivedResponseType #>> wrappedCallback = response =>
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt]: where/how is that used?

File: workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt [hunk 2]
```diff
@@ -108,13 +110,55 @@ namespace <#= qualifiedNamespace #>
...
+                    };
+
                     var ecsCommandRequestSender = entityManager.GetComponentData<CommandSenders.<#= commandDetails.CommandName #>>(entity);
                     var request = <#= commandDetails.CommandName #>.CreateRequest(entityId, payload, timeoutMillis, allowShortCircuiting, callback);
                     ecsCommandRequestSender.RequestsToSend.Add(request);
                     return request.RequestId;
                 }
 
 <# } #>
+                long ICommandRequestSender.SendCommand<TCommand, TRequest, TResponse>(CommandTypeInformation<TCommand, TRequest, TResponse> commandTypeInformation, EntityId entityId, TRequest request,
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt]: commandTypeInformation -> is that passed in so they don't need to explicitly define the generics when calling `SendCommand`?

Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt]: It also uniquely identifies the command. A (TRequest, TResponse) pair is not sufficient

Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt]: What he said

File: workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt (hunk 1 above)
Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt]: It's used to not call the thing if the sender that originally sent it is no longer valid

Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt]: but the `wrappedCallback ` is not called anywhere?

**Comment by samiwh** [comment]
Closing this as now obsolete with new changes coming soon.

**Pull request closed without merging**